SDWebImage/SDWebImagePrefetcher.m

```diff
@@ -105,8 +107,10 @@ - (nullable SDWebImagePrefetchToken *)prefetchURLs:(nullable NSArray<NSURL *> *)
             
             if (atomic_load_explicit(&(token->_finishedCount), memory_order_relaxed) + atomic_load_explicit(&(token->_skippedCount), memory_order_relaxed) >= atomic_load_explicit(&(token->_totalCount), memory_order_relaxed)) {
                 // All finished
-                [sself callCompletionBlockForToken:token];
-                [sself removeRunningToken:token];
+                if (!atomic_flag_test_and_set_explicit(&(token->_isAllFinished), memory_order_relaxed)) {
```

Comment:
Can we change `if (atomic_load_explicit(&(token->_finishedCount), memory_order_relaxed) + atomic_load_explicit(&(token->_skippedCount), memory_order_relaxed) >= atomic_load_explicit(&(token->_totalCount), memory_order_relaxed))`, the `>=` to `==` to solve this ? 
The seconds thread enter, that atomic counter may be larger than the total count, isn't it ?

Reply:
@dreampiggy , nope, let's assume all operations finished successfully, no operation error, the completion can execute multiple times, just like the demo above in my comments.